core/src/main/java/org/apache/calcite/rel/rules/AggregateExpandDistinctAggregatesRule.java

```diff
@@ -151,6 +167,19 @@ public void onMatch(RelOptRuleCall call) {
       return;
     }
 
+    // If only one distinct aggregate and one or more non-distinct aggregates,
+    // we can generate multi-phase aggregates
+    if (distinctCount == 1                // one distinct aggregate
+            && !hasGroupingSets           // no grouping sets
+            && !hasUnsupportedAggFunc     // sum/min/max/count in non-distinct aggregate
+            && nonDistinctCount > 0       // one or more non-distinct aggregates
+            && aggregate.getGroupSet().size() > 0) { // at least one group-by key present
```

Comment:
It runs into nullability issues. If no group by is present, the aggregate is not nullable. However, when we introduce the distinct column in the group by the new aggregate becomes nullable. Requiring a group by ensures that both the old and new aggregates are nullable.